teps/0061-allow-custom-task-to-be-embedded-in-pipeline.md

```diff
@@ -382,12 +424,25 @@ to verify whether the controller can handle the custom task `taskSpec`.
 How does this proposal affect the reusability, simplicity, flexibility 
 and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
 -->
+Before the implementation of this TEP, i.e. without the support for embedding a
+custom task spec in the `PipelineRun` resource, a user has to create multiple API
+requests to the Apiserver. Next, he has to ensure unique names, to avoid conflict
+in someone else's created custom task resource object. 
+
+Embedding of custom task spec avoids the problems related to name collisions
+and also improves performance by reducing the number of API requests to create custom
+task resource objects. The performance benefit, of reducing the number of API requests,
+is more evident when using web-ui based dashboard to display, pipeline details
+(e.g. in Kubeflow Pipelines with tekton as backend).
+
+Lastly, it looks aesthetically nicer and coherent with existing regular task,
+with all the custom task spec using fewer lines of yaml and all present in one place.
 
 ## Drawbacks
 
-<!--
-Why should this TEP _not_ be implemented?
--->
+One of the drawback can be, user may still use the `taskRef` for creating custom task resource,
+based pipelines. As a result, may still run into name collision issue. This can be avoided by
```

Comment:
Can you be more explicit on which `name collision` you are referring? Is it the Task name in the pipeline or the custom task resource name?

Reply:
I have tried to clarify that sentence, please take a look again.